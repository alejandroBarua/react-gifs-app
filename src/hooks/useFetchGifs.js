import {useState, useEffect} from 'react';
import { getGifs } from '../helpers/getGifs';


const useFetchGifs = (category) => {

	const [state, setState] = useState({
		data: [],
		loading: true
	});

	useEffect(() => {

		getGifs(category)
			.then(imgs => setState({ data: imgs, loading: false }));

	},[category]) // solo se renderiza el componente cuando el valor de category cambie

	return state;
}


export default useFetchGifs;