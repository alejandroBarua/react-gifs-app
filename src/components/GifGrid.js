import React from 'react';
import PropTypes from 'prop-types';

import useFetchGifs from '../hooks/useFetchGifs';
import Gif from "./Gif";

const GifGrid = ({category}) => {

	const {data : images, loading} = useFetchGifs(category);
	
	return (
		<>
			<h2 className="title-category">{category}</h2>

			{loading && <h3 className="animate__animated animate__flash">Loading...</h3> } 
			{/* // loading ? <h3>Loading...</h3> : null */}

			<div className="card-grid">
				{
					images.map(({id, ...image}) => <Gif key={id} {...image} />)
				}
			</div>
		</>
	)
}

GifGrid.propTypes = {
	category: PropTypes.string.isRequired
}

export default GifGrid;