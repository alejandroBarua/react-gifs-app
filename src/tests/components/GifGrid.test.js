import React from 'react';
import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid';
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs'); // para falsear la data

describe('gifGrid component', () => {
	
	const category = "One Punch";

	
	
	test('render component', () => {
		
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});
		
		const wrapper =  shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	})

	test('after useFetchGifs', () => {

		const gifs = [{
			id: 'ABC',
			url: 'https://localhost/anything/any.jpg',
			title: 'any'
		}]

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: true
		});
		
		const wrapper =  shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').exists()).toBe(false)
		expect(wrapper.find('Gif').length).toBe(gifs.length)

	})
})
