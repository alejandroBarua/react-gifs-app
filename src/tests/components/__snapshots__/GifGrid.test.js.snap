// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gifGrid component after useFetchGifs 1`] = `
<Fragment>
  <h2
    className="title-category"
  >
    One Punch
  </h2>
  <h3
    className="animate__animated animate__flash"
  >
    Loading...
  </h3>
  <div
    className="card-grid"
  >
    <Gif
      key="ABC"
      title="any"
      url="https://localhost/anything/any.jpg"
    />
  </div>
</Fragment>
`;

exports[`gifGrid component render component 1`] = `
<Fragment>
  <h2
    className="title-category"
  >
    One Punch
  </h2>
  <h3
    className="animate__animated animate__flash"
  >
    Loading...
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
